src/Integration/Aws/DynamoDbSession/src/SessionHandler.php

```diff
@@ -0,0 +1,206 @@
+<?php
+
+namespace AsyncAws\DynamoDbSession;
+
+use AsyncAws\DynamoDb\DynamoDbClient;
+use AsyncAws\DynamoDb\Enum\BillingMode;
+use AsyncAws\DynamoDb\Enum\KeyType;
+use AsyncAws\DynamoDb\Enum\ScalarAttributeType;
+use AsyncAws\DynamoDb\Enum\TableStatus;
+
+class SessionHandler implements \SessionHandlerInterface
+{
+    /**
+     * @var DynamoDbClient
+     */
+    private $client;
+
+    /**
+     * @var array
+     */
+    private $config;
+
+    /**
+     * @var string
+     */
+    private $dataRead = '';
+
+    /**
+     * @var string
+     */
+    private $sessionId;
+
+    /**
+     * @var string
+     */
+    private $sessionName;
+
+    /**
+     * @var bool
+     */
+    private $sessionWritten;
+
+    /**
+     * @param array{
+     *   consistent_read?: bool,
+     *   data_attribute?: string,
+     *   hash_key?: string,
+     *   session_lifetime?: int,
+     *   session_lifetime_attribute?: string,
+     *   table_name: string
+     * } $config
+     */
+    public function __construct(DynamoDbClient $client, array $config)
+    {
+        $this->client = $client;
+        $this->config = $config;
+        $this->config['data_attribute'] = $this->config['data_attribute'] ?? 'data';
+        $this->config['hash_key'] = $this->config['hash_key'] ?? 'id';
+        $this->config['session_lifetime_attribute'] = $this->config['session_lifetime_attribute'] ?? 'expires';
+    }
+
+    public function setUp(int $wait = 1): void
+    {
+        $this->client->createTable([
+            'TableName' => $this->config['table_name'],
+            'BillingMode' => BillingMode::PAY_PER_REQUEST,
+            'AttributeDefinitions' => [
+                [
+                    'AttributeName' => $this->config['hash_key'],
+                    'AttributeType' => ScalarAttributeType::S,
+                ],
+            ],
+            'KeySchema' => [
+                [
+                    'AttributeName' => $this->config['hash_key'],
+                    'KeyType' => KeyType::HASH,
+                ],
+            ],
+        ]);
+
+        do {
+            sleep($wait);
```

Comment:
No need to sleep here. You can "wait" for the table to be created. See https://github.com/async-aws/aws/blob/master/src/Integration/Laravel/Cache/src/AsyncAwsDynamoDbStore.php#L407-L443

Reply:
thanks, had been looking for this but didn't find it